homeassistant/components/media_player/plex.py

```diff
@@ -272,93 +614,316 @@ def _active_media_plexapi_type(self):
     @property
     def media_content_id(self):
         """Content ID of current playing media."""
-        if self.session is not None:
-            return self._convert_na_to_none(self.session.ratingKey)
+        return self._media_content_id
 
     @property
     def media_content_type(self):
         """Content type of current playing media."""
-        if self.session is None:
-            return None
-        media_type = self.session.type
-        if media_type == 'episode':
+        if self._session_type == 'clip':
+            _LOGGER.debug('Clip content type detected, '
+                          'compatibility may vary: %s',
+                          self.entity_id)
             return MEDIA_TYPE_TVSHOW
-        elif media_type == 'movie':
+        elif self._session_type == 'episode':
+            return MEDIA_TYPE_TVSHOW
+        elif self._session_type == 'movie':
             return MEDIA_TYPE_VIDEO
-        elif media_type == 'track':
+        elif self._session_type == 'track':
             return MEDIA_TYPE_MUSIC
-        return None
+        else:
+            return None
+
+    @property
+    def media_artist(self):
+        """Artist of current playing media, music track only."""
+        return self._media_artist
+
+    @property
+    def media_album_name(self):
+        """Album name of current playing media, music track only."""
+        return self._media_album_name
+
+    @property
+    def media_album_artist(self):
+        """Album artist of current playing media, music track only."""
+        return self._media_album_artist
+
+    @property
+    def media_track(self):
+        """Track number of current playing media, music track only."""
+        return self._media_track
 
     @property
     def media_duration(self):
         """Duration of current playing media in seconds."""
-        if self.session is not None:
-            return self._convert_na_to_none(self.session.duration)
+        return self._media_duration
+
+    @property
+    def media_position(self):
+        """Position of current playing media in seconds.
+
+        Music position appears to update slower than other media.
+        """
+        if self._session:
+            return self._media_position
+
+    @property
+    def media_position_updated_at(self):
+        """When was the position of the current playing media valid."""
+        if self._is_player_active:
+            return util.dt.utcnow()
```

Comment:
can you confirm my understanding of what you are suggesting for positioning is correct? I think you are suggesting to fake it by polling the position Initally and then increment the position based on elapsed time (no more polling the position) plus update the position when you see a play, pause, stop, next/previous track event.  I would also only set the _updated_at variable when I actually poll for position not when I fake update the position.  Finally, I can stop fake determining the position once I reach and or exceed duration.  Correct?  If so, my frozen detection feature is almost worthless as I won't be able to quickly detect a plex client who is in a playing or paused state but the position hasn't moved.  For example, start streaming from an iPhone or browser plex client, watch it show as playing in hass, and then kill that client (ex user locks phone, user closes the browser, plex client crashes, etc).  With the way I have the code now, I can know pretty quickly when that happens.  If I change it to my understanding of how you want it, I'll only know when my fake positioning goes past duration.